Switchy Client Management Screen

HestiMae opened the pull request:
Closes #25 
What's done:

- Layout of preset management and module management tabs
- Tab switcher
- Client-side method calls on SwitchyDisplayablePresets object

To-do:

- [x] Layout of data tab (covers importing and exporting)
- [x] Interface for refreshing screens using display presets
- [x] Button on quick switcher to access management screen
- [x] Refresh interface implementation for quick switcher and management screen
- [x] Use client API to send actions to the server
- [x] Lock screen after sending action until refresh
![image](https://user-images.githubusercontent.com/62225435/222058557-9356ad75-db2e-4b5e-872b-468686254e2d.png)
![image](https://user-images.githubusercontent.com/62225435/222058625-c717cda9-d07a-4ae6-bc77-1cbe15df1b21.png)

HestiMae committed:
Preset management screen with preset and module tabs visually mocked

HestiMae committed:
Merge remote-tracking branch 'origin/1.19' into 1.19

HestiMae committed:
Working confirmation dialogs

HestiMae committed:
Working implementation using API
used proper confirmation messages for dialogs
prevented misuse in delete + rename + new

HestiMae committed:
make both screens usable at once

sisby-folk commented:
Looking really good so far! I'll give this a more thorough review when you're happy with it 

https://user-images.githubusercontent.com/55819817/222342767-7374ad01-3dbe-4e0c-a214-80e340984266.mp4

HestiMae committed:
Added tooltips on module screen, bugfixes
fixed game crash on duplicate preset names
show current tab with button styling

HestiMae committed:
extracted module flow as template
changed module window from fixed size to scaling based on label size

HestiMae committed:
Merge branch 'sisby-folk:1.19' into 1.19

HestiMae committed:
Data tab template created, imports files

HestiMae committed:
First working version of import 11!!11!

HestiMae committed:
Merge branch 'sisby-folk:1.19' into 1.19

HestiMae committed:
Merge branch 'sisby-folk:1.19' into 1.19

HestiMae committed:
update to beta 5

HestiMae committed:
Merge branch 'sisby-folk:1.19' into 1.19

HestiMae committed:
Merge remote-tracking branch 'origin/1.19' into 1.19

# Conflicts:
#	build.gradle

HestiMae committed:
Merge branch 'sisby-folk:1.19' into 1.19

HestiMae committed:
fix quasi enable and disable for display

HestiMae committed:
Add export functionality!
Screen is now functionally complete! it could use some clean up, but I'm happy to merge this and then start up a cleanup PR for post-2.0 as there's no API in this thing.

HestiMae committed:
Fixing a typo

HestiMae commented:
> Screen is now functionally complete! it could use some clean up, but I'm happy to merge this and then start up a cleanup PR for post-2.0 as there's no API in this thing.

Here are a few screenshots - I'd still like to clean this up both technically and UX-wise, but it's still got an intuitive flow.

![image](https://user-images.githubusercontent.com/62225435/223952422-35692587-9023-4068-94c9-95fe7cd5beec.png)

![2023-03-09_18 36 29](https://user-images.githubusercontent.com/62225435/223952627-52206203-5727-4132-8f0e-5ff6e5a42777.png)
![2023-03-09_18 36 37](https://user-images.githubusercontent.com/62225435/223952641-2997d867-0c5c-4e75-b35d-7baab5f24f43.png)
![2023-03-09_18 36 52](https://user-images.githubusercontent.com/62225435/223952644-a6348481-b878-431e-a643-f0bcd92d8b14.png)
![2023-03-09_18 37 01](https://user-images.githubusercontent.com/62225435/223952650-00a64bae-0e12-46a8-9d4e-e03c72aefff7.png)

sisby-folk commented:
Super intense work! nice job getting it all hooked up and functional.

We've given this a play around and have a few of points of improvement - _but_ we'd like those to be optionally addressed later down the track (we'll take a crack at it if you dont) as we really want to smash UI stuff out into another submodule and get 2.0 out the door ASAP.

 - Visually I'd like more surface variety to break up some of the more complex areas. 
   - I think the "enabled disabled" module areas could be darkened so that they're "inset" into the panel.
   - Horizontal line items like modules and presets (and their buttons) could have a similar colour-change to more clearly show them as being "items"
 - Being able to rename/create multiple presets at once is no good, and it makes the behaviour of the cancel/confirm buttons strange. Limiting it to just one makes sense based on mobile UI i'm used to.
 - Reusing the module info and confirmation text for the confirmation dialog boxes is really good, but ideally we'd like to show the feedback that's coming up in the chat on the screen instead (as a toast) - I think this will tie directly into plans we have for identifying individual requests.
   - Just to quickly explain - Client API requests will gain an optional callback parameter that will be given the returned `displayPresets` alongside some kind of feedback object with an enum type and message text - you'll be able to use this for toasts. I'll do my best to adapt the screen interface to use this in some way.
